synd-contracts/src/factory/SyndicateFactory.sol

```diff
@@ -17,121 +36,261 @@ enum NamespaceState {
-contract SyndicateFactory is AccessControl, Pausable {
+/// @dev Uses UUPS proxy pattern for upgradeability and CREATE2 pattern for deterministic deployments
+contract SyndicateFactory is Initializable, AccessControlUpgradeable, PausableUpgradeable, UUPSUpgradeable {
     /// @notice Emitted when a new SyndicateSequencingChain is created
+
     event SyndicateSequencingChainCreated(
         uint256 indexed appchainId, address indexed sequencingChainAddress, address indexed permissionModuleAddress
     );
 
-    /// @notice Emitted when namespace configuration is updated
-    event NamespaceConfigUpdated(uint256 oldNamespacePrefix, uint256 newNamespacePrefix);
-
     /// @notice Emitted when a chain ID is manually marked as used
     event ChainIdManuallyMarked(uint256 indexed chainId);
 
-    bytes32 public constant MANAGER_ROLE = keccak256("MANAGER_ROLE");
+    /// @notice Emitted when a new implementation is added to allowed list
+    event ImplementationAdded(address indexed implementation);
+
+    /// @notice Emitted when a chain is banned from gas tracking
+    event ChainBannedFromGasTracking(uint256 indexed chainId, address indexed notAllowedImplementation);
+
+    /// @notice Emitted when a deterministic chainID is generated
+    event DeterministicChainIdGenerated(address indexed sender, uint256 indexed nonce, uint256 indexed chainId);
+
+    /// @notice Emitted when the creation fee is updated
+    event CreationFeeUpdated(uint256 oldFee, uint256 newFee);
 
     error ZeroAddress();
     error ChainIdAlreadyExists();
-
-    // Namespace configuration - made public for frontend access
-    uint256 public namespacePrefix;
-    uint256 public nextAutoChainId;
-    mapping(uint256 => NamespaceState) public usedNamespaces;
+    error ImplementationNotAllowed();
+    error ImplementationAlreadyAllowed();
+    error OnlyChainCanNotifyUpgrade();
+    error CannotRemoveDefaultImplementation();
+    error FailedToUpgradeToLatestImplementation();
+    error InsufficientFee();
 
     /// @notice Mapping from appchain ID to the sequencing contract address
     mapping(uint256 => address) public appchainContracts;
     uint256[] public chainIDs;
 
-    constructor(address admin) {
+    /// @notice Stub implementation for consistent proxy deployment
+    address public stubImplementation;
+
+    /// @notice Current implementation address used for new deployments
+    address public syndicateChainImpl;
+
+    /// @notice List of allowed implementation addresses for sequencing chains
+    address[] public allowedImplementations;
+    mapping(address => bool) public isImplementationAllowed;
+
+    /// @notice Chains banned from gas tracking due to not allowed implementation
+    mapping(uint256 => bool) public gasTrackingBanlist;
+    uint256 public numberOfChainsBannedFromGasTracking;
+
+    /// @notice Per-sender nonce tracking for deterministic chainID generation
```

Comment:
@gustavoguimaraes 

I think we should remove this nonce tracker and all logic related to it. just let the user specify the nonce themselves. 

They will get a chainID conflict, so they cannot re-use a nonce anyway.

Besides being simpler, the current way is a bit annoying to use.
example: User has 10 appchains deployed on chainA and  wants to move the 9th appchain to chainB - with the current code they would have to create 8 useless contracts before the nonce has reached the desired value.